frontend/employer/src/backend-api/BackendAPIProvider.tsx

```diff
@@ -0,0 +1,39 @@
+import Axios from 'axios';
+import { getBackendUrl } from 'employer/backend-api/backend-url';
+import React from 'react';
+
+import BackendAPIContext from './BackendAPIContext';
+
+const BackendAPIProvider = ({
+  children,
+}: React.PropsWithChildren<unknown>): JSX.Element => {
+  const axiosContext = React.useMemo(() => {
+    const axiosClient = Axios.create({
+      baseURL: getBackendUrl(),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    axiosClient.interceptors.request.use((config) => {
+      // Read token for anywhere, in this case directly from localStorage
+      // eslint-disable-next-line scanjs-rules/identifier_localStorage
+      const token = localStorage.getItem('token');
```

Comment:
are we planning at some point to implement and use HttpOnly cookie instead of local storage implementation? I noticed this security leak in many Hki projects.

Reply:
This is still completely open thing, I actually used this code as base: https://cretezy.com/2020/react-query-axios-authentication

Reply:
Final solution to authentication will be based to this example project (except using react-query instead of redux): https://github.com/City-of-Helsinki/example-ui-profile/tree/HP-702-two-login-types
I think thsi example has better solution for persisting token.